### ISSUE TITLE
localdispatch & deadlinedispatch not consistant

### ISSUE THREAD

**Issue by willDTF** [hypothetical-inc/GafferDeadline#88]
Hi,
i did some test using local and deadline dispatch with the same settings and rendering the exact same things
![image](https://github.com/user-attachments/assets/fd68954c-6595-41f6-a5ca-c1ad4809a1dc)

inside the box, a deadlinedispatch is used just before the render node

here are some things I've noticed 

### **using localdispatch :** 
![image](https://github.com/user-attachments/assets/755aab70-5c90-4b11-908d-73d53a1e25ef)

**the good :** 
all jobs and postprocess jobs are listed as separate jobs
dependencies are dispatched as expected
**problem :**
${script:name} is messed up. 

### **using deadlineDispatch:**
![image](https://github.com/user-attachments/assets/ae05863a-fe26-477a-a724-85e54d0642ee)
the good : 
${script:name} is fine.
**problem:** 
all jobs are packed in 2 jobs (one per renderPass). so it's hard to follow what's going on

to me, localdispatch driving deadlinedispatch is better, but ${script:name} is hardly readable
can it be fixed or it's the localdispatch that mess up the variable ?

Also, is there any example of use of DeadlineTask node  ? 

Thanks

**Comment by ericmehl** [comment]
The DeadlineTask node is kind of a specialty node for specific use cases. It submits a Deadline job with the parameters you add using the "+" button. These will be regular Deadline plugins, most likely not related to Gaffer at all. For example, you can set the `DeadlineTask.plugin` plug to `Nuke` and add a bunch of Nuke-related parameters, and when you dispatch it, it will create a native Nuke job in Deadline.

The definitive reference to what parameters should be named and what their data type is comes from Deadline's plugin installation. Each plugin will have a `.param` file which gives info for the parameters it accepts.

**Comment by willDTF** [comment]
Thank you for details, so it's now my favorite node 😁

**Comment by ericmehl** [comment]
For the job naming issue, what are your plug values for the DeadlineDispatcher and the LocalDispatcher?

For the jobs sent by DeadlineDispatch, I expect when those run, they will submit new jobs that do the actual rendering work. That's because your are dispatching twice - the first DeadlineDispatch (the one you're pushing "dispatch" on) just submits the inner dispatcher, which then submits the render and comp tasks. Does that make sense, or am I just making things muddier?

I think the typical use case for the double DeadlineDispatcher would be if you have a whole bunch of tasks being dispatched, like from a wedge node, and you don't want to wait around for that to dispatch. Adding the second dispatcher means that the first dispatcher runs on the farm and you can get tea while it runs.

**Comment by ericmehl** [comment]
> Thank you for details, so it's now my favorite node 😁

Yeah, it is quite handy considering how extensive Deadline's plugin support is for other DCCs.

**Comment by willDTF** [comment]
About job name, both have the script:name variable only, nothing more. 
Seems that in case one it's badly evaluated because of windows path being involved somewhere. So it remove backslashes and put the all path as a name

About Dispatcher, it's clear, thanks !
